js/notebook/src/plot/plotScope.js

```diff
@@ -1714,7 +1714,24 @@ define([
       }else{
         // scale only
         var level = self.zoomLevel;
-        if (my <= plotUtils.safeHeight(self.jqsvg) - self.layout.bottomLayoutMargin) {
+        var isAllPointPlots = this.stdmodel.data.length &&
+          this.stdmodel.data.every(plot => plot.type == "point");
+        if(my <= plotUtils.safeHeight(self.jqsvg) - self.layout.bottomLayoutMargin
+          && mx >= self.layout.leftLayoutMargin && isAllPointPlots) {
+            // Scrolling in the middle of the chart, autoscale Y
+            var data = this.stdmodel.data;
+            var AllYValsInRange = [].concat(
+              ...data.map(
+                d => d.elements.filter(el => el.x >= focus.xl && el.x <= focus.xr).map(el => el.y)
+              )
+            );
+            var minYValue = Math.min(...AllYValsInRange);
+            var maxYValue = Math.max(...AllYValsInRange);
+            focus.yl = minYValue;// - self.stdmodel.margin.bottom;
+            focus.yr = maxYValue;// + self.stdmodel.margin.top;
+            focus.yspan = focus.yr - focus.yl;
+        }
+        else if (my <= plotUtils.safeHeight(self.jqsvg) - self.layout.bottomLayoutMargin) {
```

Comment:
I think this `else if` should be on the previous line with the closing brace for consistency